test/Serilog.Settings.Configuration.Tests/DynamicLevelChangeTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Serilog.Core;
+using Serilog.Events;
+using Serilog.Settings.Configuration.Tests.Support;
+
+using Xunit;
+using Microsoft.Extensions.Configuration;
+
+using TestDummies.Console;
+
+namespace Serilog.Settings.Configuration.Tests
+{
+    public class DynamicLevelChangeTests
+    {
+        const string DefaultConfig = @"{
+            'Serilog': {
+                'Using': [ 'TestDummies' ],
+                'MinimumLevel': {
+                    'Default': 'Information',
+                    'Override': {
+                        'Root.Test': 'Information'
+                    }
+                },
+                'LevelSwitches': { '$mySwitch': 'Information' },
+                'WriteTo:Dummy': {
+                    'Name': 'DummyConsole',
+                    'Args': {
+                        'levelSwitch': '$mySwitch'
+                    }
+                }
+            }
+        }";
+
+        readonly ReloadableConfigurationSource _configSource;
+
+        public DynamicLevelChangeTests()
+        {
+            _configSource = new ReloadableConfigurationSource(JsonStringConfigSource.LoadData(DefaultConfig));
+        }
+
+        [Fact]
+        public void ShouldRespectDynamicLevelChanges()
+        {
+            using (var logger = new LoggerConfiguration()
+                .ReadFrom
+                .Configuration(new ConfigurationBuilder().Add(_configSource).Build())
+                .CreateLogger())
+            {
+                DummyConsoleSink.Emitted.Clear();
+                logger.Write(Some.DebugEvent());
+                Assert.Empty(DummyConsoleSink.Emitted);
+
+                DummyConsoleSink.Emitted.Clear();
+                UpdateConfig(minimumLevel: LogEventLevel.Debug);
+                logger.Write(Some.DebugEvent());
+                Assert.Empty(DummyConsoleSink.Emitted);
+
+                DummyConsoleSink.Emitted.Clear();
+                UpdateConfig(switchLevel: LogEventLevel.Debug);
+                logger.Write(Some.DebugEvent());
+                logger.ForContext(Constants.SourceContextPropertyName, "Root.Test").Write(Some.DebugEvent());
+                Assert.Single(DummyConsoleSink.Emitted);
+
+                DummyConsoleSink.Emitted.Clear();
+                UpdateConfig(overrideLevel: LogEventLevel.Debug);
+                logger.ForContext(Constants.SourceContextPropertyName, "Root.Test").Write(Some.DebugEvent());
+                Assert.Single(DummyConsoleSink.Emitted);
+            }
+
+            void UpdateConfig(LogEventLevel? minimumLevel = null, LogEventLevel? switchLevel = null, LogEventLevel? overrideLevel = null)
```

Comment:
Maybe that's personal taste, but I think turning the `UpdateConfig` Local Function into a proper method would make it clearer (and shorten the code in the test method)

Reply:
Agree, thanks.